tensorboard/webapp/widgets/line_chart_v2/lib/worker/compact_data_series.ts

```diff
@@ -0,0 +1,82 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+import {DataSeries} from '../internal_types';
+
+export interface CompactDataSeries {
+  idsAndLengths: Array<{id: string; length: number}>;
+  flattenedSeries: ArrayBuffer;
+}
+
+/**
+ * Converts array of DataSeries to a compact represenation for postMessage.
+ *
+ * Sending a large JavaScript object can lead to inefficiency as it spends significant
+ * amount of time on JSON.stringify and JSON.parse.
```

Comment:
I never actually asked, but: while this seems 100% plausible to me, I’m
curious whether you measured it and, if so, what the figures were
compared to the compaction overhead here?

Reply:
Yup, it was measured. When sending points of 1000 points (`{x: number; y: number}`), I saw about 4.5x. The testing was not extensive and does not include larger data points.